### PR TITLE
Allow DSN env variable for database connection

### PR DESCRIPTION
**Pull request by vivekhiwarkar**
### What does this PR change?

- Add support for allowing DSN string for db connection.
- Resolving this [issue](https://github.com/paralus/paralus/issues/45)

### Does the PR depend on any other PRs or Issues? If yes, please list them.

- No

### Checklist

I confirm, that I have...

- [X] Read and followed the contributing guide in `CONTRIBUTING.md`
- [ ] Added tests for this PR
- [X] Formatted the code using `go fmt` (if applicable)
- [ ] Updated [documentation on the Paralus docs site](https://github.com/paralus/website/blob/main/docs) (if applicable)
- [ ] Updated `CHANGELOG.md`

### CODE REVIEW

File: env.example [hunk 1]
```diff
@@ -10,6 +10,7 @@ DB_NAME=admindb
 DB_USER=admindbuser
 DB_PASSWORD=admindbpassword
 DB_PORT=5432 # used in docker-compose
+DSN='postgres://admindbuser:admindbpassword@localhost:5432/admindb?sslmode=disable' #default
```
Comment (meain) [inline env.example:13]:
```suggestion
DSN='postgres://admindbuser:admindbpassword@localhost:5432/admindb?sslmode=disable' # overrides DB_{ADDR,NAME,USER,PASSWORD,PORT}
```

File: main.go [hunk 2]
```diff
@@ -62,6 +62,7 @@ const (
 	dbNameEnv     = "DB_NAME"
 	dbUserEnv     = "DB_USER"
 	dbPasswordEnv = "DB_PASSWORD"
+	dbDSNEnv      = "DSN"
```
Comment (meain) [inline main.go:65]: Should we change this to `DB_DSN`? @niravparikh05 @akshay196

Comment (meain) [reply main.go:65]: We can move this to top to keep it consistent.
/nit

File: env.example (hunk 1 above)
Comment (meain) [reply env.example:13]: /nit: we can move it to top, before `DB_ADDR` to keep it consistent.

File: main.go (hunk 2 above)
Comment (niravparikh05) [reply main.go:65]: yeah I guess we could change it to DB_DSN to keep it consistent with other database attributes. However same has to be modified in helm-charts as well

**Review by meain — APPROVED** [review]
LGTM

**Pull request merged**